### ISSUE TITLE
Add support  for mongoose ^4.11.0 connections

### ISSUE THREAD

**Issue by alebabai** [balmasi/migrate-mongoose#29]
Mongoose's default connection logic is deprecated as of 4.11.0 so in here is the following warning:
```
Db.prototype.authenticate method will no longer be available in the next major release 3.x as MongoDB 3.6 will only allow auth against users in the admin db and will no longer allow multiple credentials on a socket.
Please authenticate using MongoClient.connect with auth credentials.
```
See [The `useMongoClient` Option](http://mongoosejs.com/docs/connections.html#use-mongo-client)

**Comment by dancgray** [comment]
Why not go straight to supporting mongoose ^5.1.0 ?

**Comment by balmasi** [comment]
You got it :) Released in 4.0.0
